### DISCUSSION TITLE
Can I expose a switch as a light entity in Home Assistant?

### DISCUSSION THREAD

**Discussion by ndbroadbent** [Koenkk/zigbee2mqtt#18453] [Devices]
I've got some Zigbee wall sockets that I use to control a few on/off lights. I can use the "change device type of a switch" function in Home Assistant, but I would like to be able to do this in zigbee2mqtt.

I also run into problems when I try to add both lights and switches to a Zigbee group. zigbee2mqtt creates two groups in HA (one for the switches and one for the lights.) So I want to control them all at once, I have to manually set up a group in Home Assistant. It would be great if zigbee2mqtt could support this case so that it just creates one `light` group. (Even if it's just an abstraction that still controls the two Zigbee groups behind the scenes.)

Thanks!

**Comment by francisp2** [comment]
```
` '0x00124b0024c14b04':
    friendly_name: schuindak
    homeassistant:
      switch:
        type: light
        object_id: light
      light:
        name: lichtboven_schuindak
        value_template: null
        state_value_template: '{{ value_json.state }}'
``

**Reply by ndbroadbent** [reply]
Thanks for the reply @francisp2, yes this is what I have been doing but it's very tedious to do it for all my lights. It would be great if zigbee2mqtt exposed this as an option, so I could get it to create light entities in home assistant for some of my switch modules and wall plugs.

**Reply by Danit2** [reply]
You can change the switch in homeassistant to a light. 
This works good.
